### PR TITLE
SOC-5549 Hidden spaces never found in unified search

### PR DESCRIPTION
**Pull request by nttuyen**
Solution: remove the check if space is hidden to ignore it. In the result from ES, we only get the space which current user belong to.

### CODE REVIEW

**Comment by nttuyen** [comment]
@thomasdelhomenie: I added more commit to make it return all space (except the hidden) in unified search. To do this we have to:
- Index visibility of space in ES
- Add one more condition (OR) about visibility is not hidden beside permissions

Please review again when you have time

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/search/SpaceElasticUnifiedSearchServiceConnector.java [hunk 1]
```diff
@@ -70,7 +70,24 @@ protected String getSourceFields() {
...
     return StringUtils.join(sourceFields, ",");
   }
-  
+
+  @Override
+  protected String getPermissionFilter() {
+    // Build permission base on member of space as in parent class
+    String permissions = super.getPermissionFilter();
+
+    // We want to return all space which is not hidden, that's why we want to append this into permission query
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/search/SpaceElasticUnifiedSearchServiceConnector.java]: The comment is quite misleading. I think something like "return also all non-hidden spaces, even if the user is not a member" is better. WDYT ?

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/search/SpaceIndexingServiceConnector.java [hunk 2]
```diff
@@ -117,6 +118,7 @@ public String getMapping() {
     properties.put("prettyName", postingHighlighterField);
     properties.put("displayName", postingHighlighterField);
     properties.put("description", postingHighlighterField);
+    properties.put("visibility", notAnalyzedField);
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/search/SpaceIndexingServiceConnector.java]: If we have to change the mapping, we should also take this opportunity to add the registration field IMO (even if it is not directly related to this issue). WDYT ?

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/search/SpaceElasticUnifiedSearchServiceConnector.java (hunk 1 above)
Comment (nttuyen) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/search/SpaceElasticUnifiedSearchServiceConnector.java]: Agree, the comment is updated.

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/search/SpaceIndexingServiceConnector.java (hunk 2 above)
Comment (nttuyen) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/search/SpaceIndexingServiceConnector.java]: Yes, the "registration" field is indexed into es now.

**Comment by nttuyen** [comment]
I just push one more commit to reindex space after change the visibility. 

But for the registration field I can't see any way to reindex after update it. Adding method into interface SpaceLifeCycleListener is not a good idea as it could raise issue on the old implementation of addons.
So, should we remove "registration" field from indexing or just keep it and accept the space will not be reindex after update that field for this time?

**Comment by thomasdelhomenie** [comment]
We can now use Java 8 default method to do a new method in an interface without breaking existing implementation.

**Comment by nttuyen** [comment]
I just add one more commit to reindex space when change the registration use the default method feature of java8.

**Pull request merged**